reproject/utils.py

```diff
@@ -31,6 +31,11 @@ def _dask_to_numpy_memmap(dask_array, tmp_dir):
     if isinstance(dask_array.ravel()[0].compute(), da.Array):
         dask_array = dask_array.compute()
 
+    # Cast the dask array to regular float for two reasons - first, zarr 3.0.0
+    # and later doesn't support big-endian arrays, and also we need to anyway
+    # make a native float memory mapped array below.
+    dask_array = dask_array.astype(float, copy=False)
```

Comment:
Naive question: if zarr does not support big endian arrays, shouldn't this explicitly casted to a little endian float? On s390x, `float` is big endian.